manifests/resource/upstream.pp

```diff
@@ -70,6 +70,7 @@
   concat { "${::nginx::config::conf_dir}/conf.d/${name}-upstream.conf":
     ensure => $ensure_real,
     notify => Class['::nginx::service'],
+    require => File["${::nginx::config::conf_dir}/conf.d/"]
```

Comment:
You're requiring a file here that doesn't exist. You should require `File["${::nginx::config::conf_dir}/conf.d"]`